### PR TITLE
Allow custom actions on `.sidebar-button` links.

### PR DESCRIPTION
**Pull request by nathanvda**
Not sure if this is something you would consider, but it seemed logical in our use case, instead of scattering buttons all over the map, or adding extra button-bars, why not allow to add buttons to the sidebar, that would do a custom action, instead of opening a pane. 

The change in itself is pretty small: I removed the prevention of default, and postponed till the click-handler itself, and let it check for the classes. A user still has to attach her own click-event to her custom button, but at least now that is possible.

So in short: a `li.sidebar-button` will not be handled by the click-handler, allowing the sidebar to 
contain custom buttons with custom actions.

This is working code, but I understand you would need something more to merge it: I would be happy to provide an example, add some documentation, and adapt the jquery-code to have the same feature, but I just wanted to check if this is something you would even consider.

### CODE REVIEW

**Comment by Turbo87** [comment]
could you add some example that shows how to use this?

**Comment by nathanvda** [comment]
I am trying to get it coded without using jquery, but in my code, I just add a button to the sidebar as follows: 

```html
            <ul role="tablist">
                <li><a href="#home" role="tab"><i class="fa fa-bars"></i></a></li>
                <li><a href="#profile" role="tab"><i class="fa fa-user"></i></a></li>
                <li class="disabled"><a href="#messages" role="tab"><i class="fa fa-envelope"></i></a></li>
                <li class="sidebar-button"><a href="#" data-alert="my message"><i class="fa fa-comments"></i></a></li>
            </ul>
```    

and then using jquery I attach my own click handler as follows as follows: 

```
$(a[data-alert]').on('click', function () {
   alert('something'); 
})
```

I am looking how I should convert that piece of jquery to the example using only leaflet code ;) 

So in essence: since the code allows other click-handlers, or even the default behaviour, you can add your own buttons to a sidebar which can do ... whatever ;)

**Comment by Turbo87** [comment]
thanks for reminding me: I'm aiming for having the same set of features in the leaflet version of this library and in the jQuery version, so it would be awesome if you could add the same feature to the jQuery version as well.

**Comment by SammyThomas** [comment]
This is awesome.
How can I make it not pop open the sidebar when clicked?
Thanks fellas!

File: js/leaflet-sidebar.js [hunk 1]
```diff
@@ -169,11 +168,16 @@ L.Control.Sidebar = L.Control.extend(/** @lends L.Control.Sidebar.prototype */ {
...
-            this._sidebar.close();
-        else if (!L.DomUtil.hasClass(this, 'disabled'))
-            this._sidebar.open(this.querySelector('a').hash.slice(1));
+    _onClick: function(e) {
+        if (!L.DomUtil.hasClass(this, 'sidebar-button')) {
+            e.preventDefault();
+            e.stopPropagation();
+            if (L.DomUtil.hasClass(this, 'active')) {
+                this._sidebar.close();
+            } else if (!L.DomUtil.hasClass(this, 'disabled') {
```
Comment (SammyThomas) [inline js/leaflet-sidebar.js]: Missing Closing bracket? 
"} else if (!L.DomUtil.hasClass(this, 'disabled') {"

**Comment by davidr-halomicrosystems** [comment]
Hey, I am still battling with this even after reading and attempting variations as per above.
http://stackoverflow.com/questions/37725297/leaflet-sidebar-disable-sidetab-popout
Any help would be appreciated. I would love to treat a side button without the slider coming out.

**Comment by nathanvda** [comment]
I fixed the code (the missing `)` --oooops), added it to the jquery version and added a simple button to all examples. 

I have not minified the code, is that ok?

**Comment by davidr-halomicrosystems** [comment]
Great stuff! Thanks so much!

**Comment by Turbo87** [comment]
@nathanvda in the current version of the library the event handlers are only attached when the `href` of the link starts with `#`. would that also work for your use case?

**Comment by Turbo87** [comment]
I just tried it and dropping the `href="#"` from your examples in this PR with the current state of the JS just works :)

closing this PR as there's nothing more to do

**Comment by nathanvda** [comment]
Cool, the new code looks very clean. 

I like the fact that the css-class makes my html a little more self-documenting, but dropping the `href` probably has the same effect. I see you did not add any example for this behaviour, so this is not something you want to to encourage? 

A bit unfortunate imho that the only feedback I got is "we already got a better version so closing this", but I can also understand that :+1:

**Comment by Turbo87** [comment]
> A bit unfortunate imho that the only feedback I got is "we already got a better version so closing this", but I can also understand that 👍

yeah, I'm sorry about that. I'm not using the library myself that much and haven't had a lot of time to properly act as a maintainer...

> I see you did not add any example for this behaviour, so this is not something you want to to encourage?

I didn't want to bloat the examples and depending on the context where you're using the lib (native JS, JQuery, React, Ember, ...) there might also be better alternatives.
